vector/src/main/java/im/vector/activity/AbstractWidgetActivity.kt

```diff
@@ -236,7 +242,35 @@ abstract class AbstractWidgetActivity : RiotAppCompatActivity() {
      *
      * @return true if the message is handled (it means an answer has been sent), false if not
      */
-    abstract fun dealsWithWidgetRequest(eventData: JsonDict<Any>): Boolean
+    @CallSuper
+    open fun dealsWithWidgetRequest(eventData: JsonDict<Any>): Boolean {
+        val action = eventData["action"] as String?
+
+        when (action) {
+            "integration_manager_open" -> {
+                var integType: String? = null
+                var integId: String? = null
+
+                val data = eventData["data"]
+
+                data?.let {
+                    val dict = data as JsonDict<String>
```

Comment:
It may worth to check class type in `data` values.

Reply:
Yes, I've added the checks